### PR TITLE
Dev sample updates

### PR DESCRIPTION
**Pull request by ao508**
These changes introduce the ability to submit updates for samples through the dashboard.

Features:

- Dynamically enable editable for the appropriate columns
- Cell edits are highlighted in yellow
- Summarize changes user is about to submit in a popup
- Prompt user when submitting updates and/or discarding changes
- Add controlled vocabulary for select sample metadata fields
- Added discard changes button that reverts user-made changes to previous values
- Added custom resolver that publishes sample updates to IGO_SAMPLE_UPDATE topic

General:
- Added "species" to protected fields
- cleaned up unused components/code

Co-authored by: Quan Nguyen <86090707+qu8n@users.noreply.github.com>
Signed-off-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>

### CODE REVIEW

File: frontend/src/components/UpdateModal.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        <p>Are you sure you want to submit the following changes?</p>
+        <div className="ag-theme-alpine" style={{ height: 350 }}>
+          <AgGridReact
+            rowData={rowData}
+            columnDefs={columnDefs}
+            groupRemoveSingleChildren={true}
+            autoGroupColumnDef={autoGroupColumnDef}
+            groupDefaultExpanded={1}
+          ></AgGridReact>
+        </div>
```
Comment (ao508) [inline frontend/src/components/UpdateModal.tsx:58]: on confirmation of changes, this hook should be called per sample / set of changes

https://github.com/mskcc/smile-dashboard/pull/65/files#diff-cf60dcd267836c17e6d88e62f7d84788e8bbd7bce725647d722e01df94ac638bR126

the hook is `useUpdateSamplesMutation`

example usage

```
 * const [updateSamplesMutation, { data, loading, error }] = useUpdateSamplesMutation({
 *   variables: {
 *      where: // value for 'where'
 *      update: // value for 'update'
 *      connect: // value for 'connect'
 *   },
 * });
 * 
 *```

Comment (ao508) [reply frontend/src/components/UpdateModal.tsx:58]: Example in use by test page `UpdateSamples.tsx`

```
  const [updateSamplesMutation, { ...args }] = useUpdateSamplesMutation({
    variables: {
      where: {
        smileSampleId: params.smileSampleId,
      },
      update: {
        revisable: false,
        hasMetadataSampleMetadata: [
          {
            update: {
              node: {
                cmoPatientId: "C-NEWPATIENTID",
                investigatorSampleId: "new investigator id",
                // "sampleClass": null,
                // "sampleName": null,
                // "sampleOrigin": null,
                // "sampleType": null,
                // "tumorOrNormal": null
              },
            },
          },
        ],
      },
    },
  });
```

File: graphql-server/src/index.ts [hunk 2]
```diff
@@ -156,3 +164,10 @@ async function main() {
 }
 
 main();
+function useState(arg0: never[]): [any, any] {
```
Comment (alisman) [inline graphql-server/src/index.ts:167]: interesting, what's this?

Comment (ao508) [reply graphql-server/src/index.ts:167]: Leftover, checked in by mistake

**Pull request merged**